labs/SettingsControls/src/SettingsCard/SettingsCard.cs

```diff
@@ -162,32 +159,71 @@ private void OnIsClickEnabledChanged()
         }
     }
 
-    private void OnIconChanged()
+    private void OnIsEnabledChanged(object sender, DependencyPropertyChangedEventArgs e)
+    {
+        VisualStateManager.GoToState(this, IsEnabled ? NormalState : DisabledState, true);
+    }
+
+    private void OnButtonIconChanged()
     {
-        if (self.Icon != null)
+        if (GetTemplateChild(ActionIconPresenter) is FrameworkElement buttonIconPresenter)
         {
-            VisualStateManager.GoToState(this, IconVisibleState, true);
+            buttonIconPresenter.Visibility = IsClickEnabled
+                ? Visibility.Visible
+                : Visibility.Collapsed;
         }
-        else
+    }
+
+    private void OnHeaderIconChanged()
+    {
+        if (GetTemplateChild(HeaderIconPresenter) is FrameworkElement headerIconPresenter)
         {
-            VisualStateManager.GoToState(this, IconCollapsedState, true);
+            headerIconPresenter.Visibility = HeaderIcon != null
+                ? Visibility.Visible
+                : Visibility.Collapsed;
         }
     }
 
     private void OnDescriptionChanged()
     {
-        if (self.Description != null)
+        if (GetTemplateChild(DescriptionPresenter) is FrameworkElement descriptionPresenter)
         {
-            VisualStateManager.GoToState(this, DescriptionVisibleState, true);
+            descriptionPresenter.Visibility = Description != null
+                ? Visibility.Visible
+                : Visibility.Collapsed;
         }
-        else
+    }
+
+    private void OnHeaderChanged()
+    {
+        if (GetTemplateChild(HeaderPresenter) is FrameworkElement headerPresenter)
         {
-            VisualStateManager.GoToState(this, DescriptionCollapsedState, true);
+            headerPresenter.Visibility = Header != null
+                ? Visibility.Visible
+                : Visibility.Collapsed;
         }
     }
-
-    private void OnIsEnabledChanged(object sender, DependencyPropertyChangedEventArgs e)
+    private void OnContentAlignmentChanged()
+    {
+        switch (ContentAlignment)
+        {
+            case ContentAlignment.Right: VisualStateManager.GoToState(this, RightState, true); break;
```

Comment:
If the ContentAlignment property changes after the control is laid out, this doesn't factor in the wrapping/width state.

Reply:
Done!